src/worker.cc

```diff
@@ -586,6 +586,38 @@ void parse_contours(const rapidjson::Document& doc,
   }
 }
 
+// parse all costings needed to fulfill the request, including recostings
+void parse_recostings(const rapidjson::Document& doc,
```

Comment:
before the code was avoiding to extract the costings from `recostings` due  to JSON/PBF (my guess) and preferred to parse all costing options instead. now it's parsing only the ones which are specified.